### PR TITLE
Support Snapshot lifecycle management

### PR DESCRIPTION
**Pull request by travisby**
Hello!

I want to use the snapshot lifecycle manager.  I figured I'd add a PR for it.

Unfortunately there's no support in our elastic7 library for it, so I just used the underlying client's method for it.  Let me know if this works or if we'd want to instead work upstream!

Thanks,

Travis

### CODE REVIEW

File: es/provider.go [hunk 1]
```diff
@@ -142,6 +142,7 @@ func Provider() terraform.ResourceProvider {
 			"elasticsearch_opendistro_ism_policy_mapping": resourceElasticsearchOpenDistroISMPolicyMapping(),
 			"elasticsearch_opendistro_monitor":            resourceElasticsearchOpenDistroMonitor(),
 			"elasticsearch_opendistro_roles_mapping":      resourceElasticsearchOpenDistroRolesMapping(),
+			"elasticsearch_snapshot_lifecycle_policy":     resourceElasticsearchSnapshotLifecyclePolicy(),
```
Comment (phillbaker) [inline es/provider.go:145]: This is actually an x-pack feature, can we group it under the xpack resources? We should update the filenames as well.

File: es/resource_elasticsearch_snapshot_lifecycle_policy_test.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+}
+
+var testAccElasticsearchSnapshotLifecyclePolicy = `
+resource "elasticsearch_snapshot_lifecycle_policy" "test" {
+  name = "terraform-test"
+  body = <<EOF
+{
+  "schedule": "0 30 1 * * ?", 
+  "name": "<daily-snap-{now/d}>", 
+  "repository": "my_repository",
```
Comment (phillbaker) [inline es/resource_elasticsearch_snapshot_lifecycle_policy_test.go:152]: A check for this is failing the tests https://travis-ci.org/github/phillbaker/terraform-provider-elasticsearch/jobs/692706071#L1228

We can probably create the dependent repository using, for example:

https://github.com/phillbaker/terraform-provider-elasticsearch/blob/19c1ee3dd91c35e124069549452543b410d2dc8d/es/resource_elasticsearch_snapshot_repository_test.go#L111-L118

**Comment by travisby** [comment]
(Thank you for the suggestions!  I've implemented them and am just trying to get tests passing now.  Unfortunately I can't get a good working environment to run them locally so I'm having travis iterate on them.  I'll squash when complete)

**Comment by nelg** [comment]
I also need this feature.

**Pull request merged**